src/syntax/grammar.mly

```diff
@@ -1474,6 +1474,9 @@ and parse_catch etry = parser
 	| [< '(Kwd Catch,p); '(POpen,_); name, pn = dollar_ident; s >] ->
 		match s with parser
 		| [< t,pt = parse_type_hint; '(PClose,_); e = secure_expr >] -> ((name,pn),(t,pt),e,punion p (pos e)),(pos e)
+		| [< '(PClose,_); e = secure_expr >] ->
+			let hint = (CTPath { tpackage = ["haxe"]; tname = "Exception"; tsub = None; tparams = [] },null_pos) in
```

Comment:
Doing this on the parser level looks hacky. I would prefer proper AST representation of type-less catches. For macros that would basically mean making `Catch.type` optional.

Reply:
Moved to AST.